src/participant.rs

```diff
@@ -204,6 +205,51 @@ pub(crate) trait ProtocolParticipant {
         Ok((self_initiated_outcome, is_ready))
     }
 
+    /// Process a `ready` message: tell other participants that we're ready and
+    /// see if all others have also reported that they are ready.
+    ///
+    /// XXX This version is temporary and will replace `process_ready_message`
+    /// once we've transitioned all of the subprotocols to use
+    /// `LocalStorage`.
+    fn process_ready_message_local<T: TypeTag<Value = ()>>(
+        &self,
+        message: &Message,
+        storage: &LocalStorage,
+    ) -> Result<(ProcessOutcome<Self::Output>, bool)> {
+        // Unlike in `process_ready_message`, we don't store the ready message here.
+        // That's because that would require taking `LocalStorage` as a `&mut`, which
+        // causes problems with the borrow checker when we're calling
+        // `self.process_ready_message_local(..., &mut self.local_storage)`.
+        // Once we've swapped all the protocols to use the new `LocalStorage` we'll
+        // be able to undo this limitation, since `self.storage()` will return
+        // `LocalStorage`.
```

Comment:
todo: Let's capture these tasks in an issue. Based on your order of operations, it looks like #185 will be the last protocol to update, so that could be a good home. Or a new issue.